### PR TITLE
【WIP】upgrade lockmode of partition append-only leaf childs

### PR DESCRIPTION
**Pull request by charliettxx**
related issue: https://github.com/greenplum-db/gpdb/issues/14447

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/parser/parse_clause.c [hunk 1]
```diff
@@ -344,6 +344,27 @@ setTargetTable(ParseState *pstate, RangeVar *relation,
 								   lockmode, NULL);
 	}
 
+	if (Gp_role == GP_ROLE_DISPATCH &&
+		!pstate->p_is_insert &&
+		pstate->p_target_relation->rd_rel->relkind == RELKIND_PARTITIONED_TABLE)
+	{
+		List *all_childs = find_all_inheritors(RelationGetRelid(pstate->p_target_relation), NoLock, NULL);
```
Comment (QingMa-gp) [inline src/backend/parser/parse_clause.c:351]:
```suggestion
		List *all_childs = find_all_inheritors(RelationGetRelid(pstate->p_target_relation), NoLock, NULL);
```

File: src/backend/parser/parse_clause.c [hunk 2]
```diff
@@ -344,6 +344,27 @@ setTargetTable(ParseState *pstate, RangeVar *relation,
...
+
+			Relation rel = CdbTryOpenTable(childoid, lockmode, &childLockUpgraded);
+
+			/*
+			 * Drop the rel refcount, but keep the access lock till end of transaction
+			 */
+			table_close(rel, NoLock);
+		}
+	}
+
```
Comment (QingMa-gp) [inline src/backend/parser/parse_clause.c:367]:
```suggestion
	if (Gp_role == GP_ROLE_DISPATCH &&
		pstate->p_target_relation->rd_rel->relkind == RELKIND_PARTITIONED_TABLE)
	{
		List *all_childs = find_all_inheritors(RelationGetRelid(pstate->p_target_relation), NoLock, NULL);

		ListCell *lc = NULL;
		bool childLockUpgraded = false;
		foreach(lc, all_childs)
		{
			Oid childoid = lfirst_oid(lc);

			Relation rel = CdbTryOpenTable(childoid, lockmode, &childLockUpgraded);

			/*
			 * Drop the rel refcount, but keep the access lock till end of transaction
			 */
			table_close(rel, NoLock);
		}
	}

```

**Review by charliettxx — COMMENTED** [review]
!!!!!

File: src/backend/parser/parse_clause.c
Comment (charliettxx) [inline src/backend/parser/parse_clause.c:1]: delete?

**Pull request closed without merging**